### PR TITLE
lint/contrib/doc updates

### PR DESCRIPTION
**Pull request by jonatack**

### CODE REVIEW

**Comment by fanquake** [comment]
> Fix the following linter errors observed

There are currently no linter errors if you [run the linter using the container](https://github.com/bitcoin/bitcoin/tree/master/test/lint), which installs the same versions as the CI. This is the recommended way to run the linters, as it avoids differences that may occur due to different versions of dependencies installed on your local system, and the need for changes like this.

> since recent releases of lief

As LIEF mostly exists here for the type stubs, I don't think there's a reason to diverge it from the version we use in Guix (0.13.2), if things are working fine.

**Comment by fanquake** [comment]
> This patch also avoids some duplicate work by the person who updates lief later on, who will most likely not see this if it is binned.

If someone if updating LIEF in Guix, they need to check the changes, and update the scripts for the new version, if needed.


I'm guessing your changes here will actually just break the Guix build, given it's still using 0.13.2?

**Comment by fanquake** [comment]
> I'm guessing your changes here will actually just break the Guix build, given it's still using 0.13.2?

```bash
time BASE_CACHE="/base_cache" SOURCES_PATH="/sources" SDK_PATH="/SDKs" ./contrib/guix/guix-build
< snip >
make[1]: Leaving directory '/distsrc-base/distsrc-950b3d4587f0-x86_64-linux-gnu'
Traceback (most recent call last):
  File "/distsrc-base/distsrc-950b3d4587f0-x86_64-linux-gnu/./contrib/devtools/security-check.py", line 230, in <module>
    lief.Binary.FORMATS.ELF: {
AttributeError: type object 'lief._lief.Binary' has no attribute 'FORMATS'
F
======================================================================
FAIL: test_ELF (__main__.TestSecurityChecks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/distsrc-base/distsrc-950b3d4587f0-x86_64-linux-gnu/./contrib/devtools/test-security-check.py", line 60, in test_ELF
    self.assertEqual(call_security_check(cc, source, executable, ['-Wl,-zexecstack','-fno-stack-protector','-Wl,-znorelro','-no-pie','-fno-PIE', '-Wl,-z,separate-code']),
AssertionError: Tuples differ: (1, '') != (1, 'test1: failed PIE NX RELRO Canary CONTROL_FLOW')

First differing element 1:
''
'test1: failed PIE NX RELRO Canary CONTROL_FLOW'

- (1, '')
+ (1, 'test1: failed PIE NX RELRO Canary CONTROL_FLOW')

----------------------------------------------------------------------
Ran 1 test in 0.086s

FAILED (failures=1)
make: *** [Makefile:1355: test-security-check] Error 1
```

**Comment by fanquake** [comment]
> would you be concept ACK on updating the guix manifest to a recent lief (0.14 or maybe imminent 0.15), or prefer to hold off for now?

I don't really think we need to update or change anything in our release infrastructure right now, particularly if the only reason is so that linters can be run against versions of dependencies that we don't use.

If anything, I think the docs could be updated in some way to say that running the linters outside the container, or with versions that differ from the CI is generally unsupported.

**Comment by tdb3** [comment]
@jonatack @kevkevinpal, what are your thoughts on reopening PR #30106 and dropping the test runner `insuffient` spelling error from 30084?

Seems like this PR (30084) is scoped to more than just the spelling error and fixing the error could help prevent near term lint issues for open PRs.

Maybe I'm missing something and this was already addressed?

File: contrib/devtools/security-check.py [hunk 1]
```diff
@@ -250,6 +250,8 @@ def check_MACHO_branch_protection(binary) -> bool:
     for filename in sys.argv[1:]:
         try:
             binary = lief.parse(filename)
+            if binary is None:
+                continue
```
[the remark this answers is not in the document]
Comment (fanquake) [reply contrib/devtools/security-check.py:254]: I agree, this change doesn't look like an improvement.

**Comment by fanquake** [comment]
Moved to draft for now, given it's not clear what the status of this is.

**Pull request closed without merging**